src/NUnitFramework/tests/Internal/Filters/NotFilterTests.cs

```diff
@@ -67,7 +67,7 @@ public void ExplicitMatchTest()
         {
             var filter = new NotFilter(new CategoryFilter("Dummy"));
 
-            Assert.False(filter.IsExplicitMatch(_topLevelSuite));
+            Assert.True(filter.IsExplicitMatch(_topLevelSuite));
```

Comment:
I actually checked, and CategoryExpression  does not appear to be referenced, or even included, in the current solution.  So it did not seem relevant.

 

However, I did examine the implication of NotFilter being applied to other then just a CategoryFilter, and the results seemed to be the same.

 

From: CharliePoole [mailto:notifications@github.com] 
Sent: Saturday, October 03, 2015 11:59 PM
To: nunit/nunit
Cc: Neil Colvin
Subject: Re: [nunit] Fix NotFilter so -exclude works correctly (#862)

 

In src/NUnitFramework/tests/Internal/Filters/NotFilterTests.cs <https://github.com/nunit/nunit/pull/862#discussion_r41094229> :

> @@ -67,7 +67,7 @@ public void ExplicitMatchTest()
>          {
>              var filter = new NotFilter(new CategoryFilter("Dummy"));
>  
> -            Assert.False(filter.IsExplicitMatch(_topLevelSuite));
> +            Assert.True(filter.IsExplicitMatch(_topLevelSuite));

If you feel like adding a few tests, we could try to correct this. Otherwise, I'd say leave it for another issue, especially since #374 <https://github.com/nunit/nunit/issues/374>  is likely to get us back into this code anyway.

—
Reply to this email directly or view it on GitHub <https://github.com/nunit/nunit/pull/862/files#r41094229> .  <https://github.com/notifications/beacon/AFh64VowbRfR7PWkyWkOAxnpS2vOoFK8ks5o4JuPgaJpZM4GIcgJ.gif>